app/vmselect/netstorage/tenant_cache.go

```diff
@@ -195,3 +215,37 @@ func alignTrToDay(tr *storage.TimeRange) {
 func hasIntersection(a, b storage.TimeRange) bool {
 	return a.MinTimestamp <= b.MaxTimestamp && a.MaxTimestamp >= b.MinTimestamp
 }
+
+// isFullyCovered checks if the tr is fully covered by entries
+// Expects that tr is aligned to the day boundaries
+func isFullyCovered(tr storage.TimeRange, entries []storage.TimeRange) bool {
```

Comment:
This function looks over-engineered to me. It's better to replace `hasIntersection` function with check if requested time-range with-in any cached elements. It's cheap and easy to understand.

Reply:
PTAL - https://github.com/VictoriaMetrics/VictoriaMetrics/pull/9079/commits/283cd27e99f32bc478a0383afb6f122bb2694bed